scripts/o3de/o3de/enable_gem.py

```diff
@@ -102,6 +105,27 @@ def enable_gem_in_project(gem_name: str = None,
     # Convert each path to pathlib.Path object and filter out duplicates using dict.fromkeys
     buildable_gems = list(dict.fromkeys(map(lambda gem_path_string: pathlib.Path(gem_path_string), buildable_gems)))
 
+    # check compatibility
+    if force:
+        logger.warning(f'Bypassing version compatibility check for {gem_json_data["gem_name"]}.')
+    else:
+        incompatible_objects = compatibility.get_gem_project_incompatible_objects(gem_json_data, project_path, check=check, gem_paths=buildable_gems)
+        if check:
+            if incompatible_objects:
+                logger.info(f'{gem_json_data["gem_name"]} is not known compatible with the '
+                    'following objects/APIs and requires the --force parameter to register:')
+                for element in incompatible_objects:
+                    logger.info(f'  {element}')
+            else:
+                logger.info(f'{gem_json_data["gem_name"]} is compatible with this project')
+            return 0
```

Comment:
I say we should still return non-zero if the check option finds incompatible objects.

That way it can be used to chain operations in shell scripting.
i.e
```bash
~/o3de$ if scripts/o3de.sh enable-gem --check -gn Multiplayer; then
     # do stuff
fi
```